### PR TITLE
Feature/single recipient kid support

### PR DESCRIPTION
**Pull request by lusis**
Okay I've gotten the tests written. I understand if you want to change/leave this out a bit but I think even with the possible new Recipient interface, a top-level Kid is still supported iirc so it's not wasted work.

### CODE REVIEW

File: jwe.go [hunk 1]
```diff
@@ -56,6 +56,15 @@ type recipientInfo struct {
 	encryptedKey []byte
 }
 
+// GetKid retrieces the (optional) kid attached to the object
```
Comment (alokmenghrajani) [inline jwe.go]: s/retrieces/retrieves/

**Comment by csstaub** [comment]
Sorry, but I don't think we should merge this to master in the current state. Due to the way Go does packaging I want to keep master as backwards-compatible as possible at all times, so we wouldn't want to merge an API that we don't want to support long-term. We should come up with a long-term solution for setting headers and then stick to that. I can however merge this to a non-master branch if you wish.

**Comment by csstaub** [comment]
See https://github.com/square/go-jose/pull/58 for an alternate solution

**Comment by zamicol** [comment]
Should this be rejected thanks to #58?

**Pull request closed without merging**